pgjdbc/src/main/java/org/postgresql/core/BaseConnection.java

```diff
@@ -64,19 +66,86 @@ ResultSet execSQLQuery(String s, int resultSetType, int resultSetConcurrency)
   ReplicationProtocol getReplicationProtocol();
 
   /**
-   * <p>Construct and return an appropriate object for the given type and value. This only considers
-   * the types registered via {@link org.postgresql.PGConnection#addDataType(String, Class)} and
-   * {@link org.postgresql.PGConnection#addDataType(String, String)}.</p>
+   * {@inheritDoc}
+   *
+   * <p>
+   * Performs a defensive copy, per specification that returned map should be
+   * modifiable.  Use {@link #getTypeMapNoCopy()} to avoid this defensive copy.
+   * </p>
+   *
+   * @see  #getTypeMapNoCopy()
+   */
+  @Override
+  Map<String, Class<?>> getTypeMap() throws SQLException;
+
+  /**
+   * Gets the unmodifiable type map for this connection.
+   * This will likely perform better than {@link #getTypeMap()}
+   * because no defensive copy is made.
+   */
+  Map<String, Class<?>> getTypeMapNoCopy();
+
+  /**
+   * Gets the unmodifiable map used for type inference of custom types.
+   * This is required because the server returns base types for domains.
```

Comment:
I'm afraid the description does not clarify what is stored in the map.

Would you please elaborate?

Reply:
We have reworked this part to be less about "inference map" and more about "inverted map", which better represents what it is.  Please see commit for more details: https://github.com/pgjdbc/pgjdbc/pull/1378/commits/cd9da18e04675dab24765109682d6f8c7cbda011